### ISSUE TITLE
Metadata version mismatch for module

### ISSUE THREAD

**Issue by shubhamagrawal280395** [georgipeltekov/ngx-file-drop#27]
Hello,
I am using angular 4 and after following demo it shows following error when I compile my project

ERROR in Error: Metadata version mismatch for module C:/Projects/plugins_dev/IDPortrait/ExtensionContent/angularIDPortrait/node_modules/ngx-file-drop/node_modules/@angular/core/core.d.ts, found version 4, expected 3, resolving symbol FileDropModule in C:/Projects/plugins_dev/IDPortrait/ExtensionContent/angularIDPortrait/node_modules/ngx-file-drop/ngx-file-drop.d.ts, resolving symbol FileDropModule in C:/Projects/plugins_dev/IDPortrait/ExtensionContent/angularIDPortrait/node_modules/ngx-file-drop/ngx-file-drop.d.ts
    at syntaxError (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler\bundles\compiler.umd.js:1729:34)
    at simplifyInContext (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler\bundles\compiler.umd.js:24979:23)
    at StaticReflector.simplify (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler\bundles\compiler.umd.js:24991:13)
    at StaticReflector.annotations (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler\bundles\compiler.umd.js:24418:41)
    at _getNgModuleMetadata (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler-cli\src\ngtools_impl.js:138:31)
    at _extractLazyRoutesFromStaticModule (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler-cli\src\ngtools_impl.js:109:26)
    at C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler-cli\src\ngtools_impl.js:129:27
    at Array.reduce (native)
    at _extractLazyRoutesFromStaticModule (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler-cli\src\ngtools_impl.js:128:10)
    at Object.listLazyRoutesOfModule (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler-cli\src\ngtools_impl.js:53:22)
    at Function.NgTools_InternalApi_NG_2.listLazyRoutes (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@angular\compiler-cli\src\ngtools_api.js:91:39)
    at AotPlugin._getLazyRoutesFromNgtools (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@ngtools\webpack\src\plugin.js:212:44)
    at _donePromise.Promise.resolve.then.then.then.then.then (C:\Projects\plugins_dev\IDPortrait\ExtensionContent\angularIDPortrait\node_modules\@ngtools\webpack\src\plugin.js:448:24)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)

**Comment by georgipeltekov** [comment]
Could you please share you package.json file?

**Comment by andrei-sadagurschi** [comment]
I am dealing with the same exact problem when building with AOT.
Angular version 4.4.6.
npm version 5.5.1.
node version 8.9.0.

**Comment by georgipeltekov** [comment]
can you give more details about the package.json file. @maartenmensink  fyi

**Comment by andrei-sadagurschi** [comment]
Here is the dependencies part:

```
"dependencies": {
    "@angular/animations": "^4.4.6",
    "@angular/common": "^4.4.6",
    "@angular/compiler": "^4.4.6",
    "@angular/compiler-cli": "^4.4.6",
    "@angular/core": "^4.4.6",
    "@angular/forms": "^4.4.6",
    "@angular/http": "^4.4.6",
    "@angular/platform-browser": "^4.4.6",
    "@angular/platform-browser-dynamic": "^4.4.6",
    "@angular/platform-server": "^4.4.6",
    "@angular/router": "^4.4.6",
    "@angularclass/conventions-loader": "^1.0.13",
    "@angularclass/hmr": "^1.2.2",
    "@angularclass/hmr-loader": "~3.0.2",
    "@ng-bootstrap/ng-bootstrap": "^1.0.0-beta.5",
    "@types/hammerjs": "^2.0.34",
    "@types/jasmine": "^2.5.47",
    "@types/node": "^7.0.18",
    "@types/selenium-webdriver": "~3.0.3",
    "@types/source-map": "^0.5.0",
    "@types/uglify-js": "^2.6.29",
    "@types/webpack": "^2.2.15",
    "add-asset-html-webpack-plugin": "^2.0.1",
    "angular-2-dropdown-multiselect": "^1.6.0",
    "angular2-csv": "^0.2.5",
    "angular2-template-loader": "^0.6.2",
    "assets-webpack-plugin": "^3.5.1",
    "awesome-typescript-loader": "~3.1.3",
    "bootstrap": "^4.0.0-beta",
    "chart.js": "^2.5.0",
    "codelyzer": "~3.0.1",
    "copy-webpack-plugin": "^4.0.1",
    "core-js": "^2.4.1",
    "css-loader": "^0.28.1",
    "exports-loader": "^0.6.4",
    "expose-loader": "^0.7.3",
    "extract-text-webpack-plugin": "~2.1.0",
    "file-loader": "^0.11.1",
    "find-root": "^1.0.0",
    "font-awesome": "^4.7.0",
    "gh-pages": "^1.0.0",
    "html-webpack-plugin": "^2.28.0",
    "http-server": "^0.10.0",
    "ie-shim": "^0.1.0",
    "imports-loader": "^0.7.1",
    "istanbul-instrumenter-loader": "2.0.0",
    "jasmine-core": "^2.6.1",
    "jquery": "^3.2.1",
    "json-loader": "^0.5.4",
    "karma": "^1.7.0",
    "karma-chrome-launcher": "^2.1.1",
    "karma-coverage": "^1.1.1",
    "karma-jasmine": "^1.1.0",
    "karma-mocha-reporter": "^2.2.3",
    "karma-remap-coverage": "^0.1.4",
    "karma-sourcemap-loader": "^0.3.7",
    "karma-webpack": "2.0.3",
    "ng-router-loader": "^2.1.0",
    "ng2-charts": "^1.5.0",
    "ng2-currency-mask": "^4.3.4",
    "ng2-google-charts": "^2.3.0",
    "ng2-toastr": "^4.0.1",
    "ng2-validation": "^4.2.0",
    "ngc-webpack": "2.0.0",
    "ngx-file-drop": "^2.0.0",
    "ngx-loading": "^1.0.7",
    "ngx-pagination": "3.0.1",
    "node-sass": "^4.5.2",
    "npm-run-all": "^4.0.2",
    "optimize-js-plugin": "0.0.4",
    "parse5": "^3.0.2",
    "popper.js": "^1.12.5",
    "protractor": "^5.1.1",
    "raw-loader": "0.5.1",
    "reflect-metadata": "^0.1.10",
    "rimraf": "~2.6.1",
    "rxjs": "^5.4.3",
    "sass-loader": "^6.0.5",
    "script-ext-html-webpack-plugin": "^1.8.0",
    "source-map-loader": "^0.2.1",
    "string-replace-loader": "1.2.0",
    "style-loader": "^0.17.0",
    "to-string-loader": "^1.1.5",
    "ts-node": "^3.0.4",
    "tslib": "^1.7.0",
    "tslint": "^5.2.0",
    "tslint-loader": "^3.5.3",
    "typedoc": "^0.7.0",
    "typescript": "^2.3.2",
    "url-loader": "^0.5.8",
    "webpack": "^2.5.1",
    "webpack-dev-middleware": "^1.10.2",
    "webpack-dev-server": "2.4.5",
    "webpack-dll-bundles-plugin": "^1.0.0-beta.5",
    "webpack-merge": "~4.1.0",
    "zone.js": "^0.8.10"
  },
  "devDependencies": {},
```

**Comment by andrei-sadagurschi** [comment]
For the scripts part I have:

```
"build:aot:qa": "npm run clean:lock && npm run clean:dist && npm run clean:aot && webpack --config config/webpack.qa.js --bail && cd dist && zip -r ../dist_qa.zip *",
"clean:lock": "npm run rimraf -- package-lock.json",
"clean:dist": "npm run rimraf -- dist",
"clean:aot": "npm run rimraf -- compiled",
```

The problem occurs when trying to build with
`npm run build:aot:qa`

**Comment by georgipeltekov** [comment]
can you try with version 2.0.1

**Comment by andrei-sadagurschi** [comment]
I am reproducing the problem with version 2.0.1.

**Comment by andrei-sadagurschi** [comment]
I've also tried older versions. Same problem.

**Comment by maartenmensink** [comment]
Please provide a repo that reproduces the error. The current package works with angular-cli build so we will need to find out the difference between  the builds.

**Comment by andrei-sadagurschi** [comment]
I cannot provide a repo at the moment, but here is another project with the same problem that got fixed as the owner says. Maybe it helps.

**Comment by andrei-sadagurschi** [comment]
You can also check the commits between version 3.0.2 and 3.0.3 of this project: https://github.com/michaelbromley/ngx-pagination/commits/master
They are fixing the same exact problem.

**Comment by maartenmensink** [comment]
@andrei-sadagurschi thanks, i will have a look and try if i can reproduce the issue.

**Comment by maartenmensink** [comment]
I have created a project([repo](https://github.com/SpringIT/ngx-file-drop-test)) based on angular 4.4.6. I am unable to reproduce the error with either builds.

I have tested with 
```
ng build
ng build --prod
ng build --prod --aot=true
```
I read the provided links and read about the solutions provided by others. Most of them subscribe that there is incorrect package version is installed of one of the angular packages. So without any futher information its hard to reproduce this issue.

If you find the opportunity to create a repo which reproduces the issue is hard for me to debug this.
